File: L7GoogleEasy/LongestConsecutiveSequence.java

import java.util.HashSet;
public class LongestConsecutiveSequence {
    public int longestConsecutive(int[] nums) {
        HashSet<Integer> set = new HashSet<>();
        for (int num : nums) {
            set.add(num);
        }
        int longest = 0;
        for (int num : nums) {
            if (!set.contains(num - 1)) {
                int current = num;
                int count = 1;
                while (set.contains(current + 1)) {
                    current++;
                    count++;
                }
                longest = Math.max(longest, count);
            }
        }
        return longest;
    }
    
    public static void main(String[] args) {
        LongestConsecutiveSequence lcs = new LongestConsecutiveSequence();
        int[] nums = { 100, 4, 200, 1, 3, 2 };
        System.out.println(lcs.longestConsecutive(nums)); // Output: 4

        int[] nums2 = { 0, 3, 7, 2, 5, 8, 4, 6, 9, 1 };
        System.out.println(lcs.longestConsecutive(nums2)); // Output: 9

        int[] nums3 = { -1, -2, -3 };
        System.out.println(lcs.longestConsecutive(nums3)); // Output: 3
    }
        
}

File: L7GoogleEasy/LongestConsecutiveSequence.cpp
#include <iostream>
#include <unordered_set>
#include <vector>
using namespace std;

class LongestConsecutiveSequence
{
public:
    int longestConsecutive(vector<int> &nums)
    {
        unordered_set<int> set(nums.begin(), nums.end());
        int longest = 0;

        for (int num : nums)
        {
            // Check if it's the start of a sequence
            if (set.find(num - 1) == set.end())
            {
                int current = num;
                int count = 1;

                // Count the length of the sequence
                while (set.find(current + 1) != set.end())
                {
                    current++;
                    count++;
                }

                longest = max(longest, count);
            }
        }

        return longest;
    }
};

int main()
{
    LongestConsecutiveSequence lcs;

    // Test case 1
    vector<int> nums1 = {100, 4, 200, 1, 3, 2};
    cout << lcs.longestConsecutive(nums1) << endl; // Output: 4

    // Test case 2
    vector<int> nums2 = {0, 3, 7, 2, 5, 8, 4, 6, 0, 1};
    cout << lcs.longestConsecutive(nums2) << endl; // Output: 9

    // Test case 3
    vector<int> nums3 = {-1, -2, -3};
    cout << lcs.longestConsecutive(nums3) << endl; // Output: 3

    return 0;
}
